Feature/different client types

thorsten-wolf-neptune opened the pull request:
As mentioned https://github.com/magieno/sqlite-client/issues/2 i want to suggest this adjustment to allow different client types that allow different storage variations:  OPFS, OPFS SAH, MEMORY.
Feel free to change/adjust/give feedback :-)

thorsten-wolf-neptune committed:
wip

thorsten-wolf-neptune committed:
wip

thorsten-wolf-neptune committed:
all options

thorsten-wolf-neptune committed:
make exec parameters optional

etiennenoel commented:
High level feedback until I do a proper deep dive:

* Can you split every class/enum/interface into their own individual files
* Can you have the enums start with a capital letter and their value be all in caps (I think most of it are)

I will continue to review once I have a bit more time, thanks!

thorsten-wolf-neptune committed:
implemented feedback

thorsten-wolf-neptune commented:
I implemented your feedback. 
Additional Notes:
- now the memory_worker and memory_main are dedicated clientTypes
- i grouped all SqliteClients now in a "clients" subfolder to have a better structure.
- the package now returns an index.js (that internally exports the other necessary files for the outside)
- i checked the enums to align with your guidelines. Only exception is the ExecuteSqlRowMode and ExecuteSqlReturnValue because the values need to match the actual sqlite parameter values. Maybe we don't need them and sqlite has some type definitions we can reuse?
- I have two '// todo ' comments where i added "as any" because my typescript knowledge is too limited to get that right ;-)

I also adjusted the demo fork:
![image](https://github.com/magieno/sqlite-client/assets/83953375/44ca03b5-8392-4ed9-9774-6a4fdac8a5ac)
